shaders/src/main/glsl/samples/310es/stable_maze.frag

```diff
@@ -0,0 +1,102 @@
+#version 310 es
+
+precision highp float;
+
+layout(location = 0) out vec4 _GLF_color;
+uniform vec2 resolution;
+
+int map[16*16];
+
+void main()
+{
+    vec2 pos = gl_FragCoord.xy / resolution;
+	ivec2 ipos = ivec2(int(pos.x*16.0),int(pos.y*16.0));
+
+	int i;
+	for (i = 0; i < 16*16; i++)
+		map[i] = 0;
+
+	ivec2 p = ivec2(0, 0);
+	bool canwalk = true;
+	int v = 0;
+	do
+	{
+		v++;
+		int directions = 0;
+
+		if (p.x > 0 && map[(p.x - 2) + (p.y) * 16] == 0)
+			directions += 1;
+		if (p.y > 0 && map[(p.x) + (p.y - 2) * 16] == 0)
+			directions += 1;
+		if (p.x < 14 && map[(p.x + 2) + (p.y) * 16] == 0)
+			directions += 1;
+		if (p.y < 14 && map[(p.x) + (p.y + 2) * 16] == 0)
+			directions += 1;
+
+		if (directions == 0)
+		{
+			canwalk = false;
+			int j;
+			for (i = 0; i < 8; i++)
+			{
+				for (j = 0; j < 8; j++)
+				{
+					if (map[j*2+i*2*16] == 0)
+					{
+						p.x = j * 2;
+						p.y = i * 2;
+						canwalk = true;
+					}
+				}
+			}			
+			map[(p.x) + (p.y) * 16] = 1;
+		}
+		else
+		{
+			int d = v % directions;
+			v += directions;
+
+			if (d >= 0 && p.x > 0 && map[(p.x - 2) + (p.y) * 16] == 0)
+			{
+				d--;
+				map[(p.x) + (p.y) * 16] = 1;
+				map[(p.x - 1) + (p.y) * 16] = 1;
+				map[(p.x - 2) + (p.y) * 16] = 1;
+				p.x -= 2;
+			}
+			if (d >= 0 && p.y > 0 && map[(p.x) + (p.y - 2) * 16] == 0)
+			{
+				d--;
+				map[(p.x) + (p.y) * 16] = 1;
+				map[(p.x) + (p.y - 1) * 16] = 1;
+				map[(p.x) + (p.y - 2) * 16] = 1;
+				p.y -= 2;
+			}
+			if (d >= 0 && p.x < 14 && map[(p.x + 2) + (p.y) * 16] == 0)
+			{
+				d--;
+				map[(p.x) + (p.y) * 16] = 1;
+				map[(p.x + 1) + (p.y) * 16] = 1;
+				map[(p.x + 2) + (p.y) * 16] = 1;
+				p.x += 2;
+			}
+			if (d >= 0 && p.y < 14 && map[(p.x) + (p.y + 2) * 16] == 0)
+			{
+				d--;
+				map[(p.x) + (p.y) * 16] = 1;
+				map[(p.x) + (p.y + 1) * 16] = 1;
+				map[(p.x) + (p.y + 2) * 16] = 1;
+				p.y += 2;
+			}
+		}
+		if (map[ipos.y*16+ipos.x] == 1)
+		{
+			_GLF_color = vec4(1.0,1.0,1.0,1.0);
+			return;
+		}
+
+	} 
+	while (canwalk);
+	
+	_GLF_color = vec4(0.0,0.0,0.0,1.0);

```

Comment:
I love the maze image that this produces!

Reply:
Different patterns can be generated by changing the initial marching position (ivec2 p).